app/assets/javascripts/samples/sample_datatable.js

```diff
@@ -91,23 +96,65 @@ table = $("#samples").DataTable({
         }
     },
     stateSaveCallback: function (settings, data) {
-        // Set a cookie with the table state using the organization id
-        localStorage.setItem('datatables_state/' + $("#samples").attr("data-organization-id"), JSON.stringify(data));
+        // Send an Ajax request to the server with the state object
+        var org = $("#samples").attr("data-organization-id")
+        var user = $("#samples").attr("data-user-id")
+
+        // Save correct data
+        if (loadFirstTime == true) {
+          data = myData;
+        }
+
+        $.ajax( {
+          url: '/state_save/'+org+'/'+user,
+          data: {org: org, state: data},
+          dataType: "json",
+          type: "POST",
+          success: function () {},
+          error: function () {}
+        } );
+        loadFirstTime = false;
     },
     stateLoadCallback: function (settings) {
-        // Load the table state for the current organization
-        var state = localStorage.getItem('datatables_state/' + $("#samples").attr("data-organization-id"));
-        if (state !== null) {
-           return JSON.parse(state);
-        }
-        return null;
+        var o;
+        // Send an Ajax request to the server to get the data. Note that
+        // this is a synchronous request since the data is expected back from the
+        // function
+        var org = $("#samples").attr("data-organization-id")
+        var user = $("#samples").attr("data-user-id")
+
+        $.ajax( {
+          url: '/state_load/'+org+'/'+user,
+          data: {org: org},
+          async: false,
+          dataType: "json",
+          type: "POST",
+          success: function (json) {
+            o = json.state;
```

Comment:
Also, use newer Ajax syntax: change to `stateLoadCallback: function (settings, callback) {` and just add here `callback( json );` and delete `return o;`.